components/crossplane-config/base/rbac.yaml

```diff
@@ -0,0 +1,35 @@
+---
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: crossplane-sa
+  namespace: crossplane-system
+
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: crossplane-clusterrole
+rules:
+- apiGroups:
+  - ""
+  resources:
+  - secrets
+  verbs:
+  - "*"
+
+---
+# Allow the crossplane-clusterrole SA in the clusters namespace to manage clusters
+apiVersion: rbac.authorization.k8s.io/v1
+kind: RoleBinding
+metadata:
+  name: crossplane-rb
+  namespace: crossplane-system
+subjects:
+- kind: ServiceAccount
+  name: crossplane-sa
+  namespace: crossplane-system
+roleRef:
+  kind: ClusterRole
+  name: crossplane-clusterrole
+  apiGroup: rbac.authorization.k8s.io
```

Comment:
This only grants `crossplane-sa` the ability to manage secrets within the `crossplane-system` namespace. Instead, this SA needs to be able to create, at a minimum, the following resources across the entire cluster:

- namespaces
- serviceaccounts
- secrets
- roles
- rolebindings
- resourcequotas
- clusterresourcequotas
- networkpolicies

List adapted from the manifests in https://github.com/redhat-appstudio/infra-deployments/tree/main/components/sandbox/tiers/src/appstudio-env.